From f5f88e39822de17e264baffbda162c0788ceb98f Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 13 Feb 2018 09:33:43 -0800
Subject: [PATCH 0272/1851] dt-bindings: display: xlnx: zynqmp-dpsub: Make
 layers as child node

commit d3e7b682202773b5818e52f5bdc9feb846648c92 from
https://github.com/Xilinx/linux-xlnx.git

This change is needed for Xilinx bridge support as each bridge
requires a device node, thus revert it back.

The change will have to submitted along with Xilinx bridge
patch set.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/display/xlnx/xlnx,zynqmp-dpsub.txt  | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/Documentation/devicetree/bindings/display/xlnx/xlnx,zynqmp-dpsub.txt b/Documentation/devicetree/bindings/display/xlnx/xlnx,zynqmp-dpsub.txt
index 7524cce299b1..46d0c7671ee5 100644
--- a/Documentation/devicetree/bindings/display/xlnx/xlnx,zynqmp-dpsub.txt
+++ b/Documentation/devicetree/bindings/display/xlnx/xlnx,zynqmp-dpsub.txt
@@ -27,6 +27,10 @@ Required properties:
 
 - power-domains: phandle for the corresponding power domain
 
+- vid-layer, gfx-layer: Required to represent available layers
+
+Required layer properties
+
 - dmas: phandles for DMA channels as defined in
   Documentation/devicetree/bindings/dma/dma.txt.
 - dma-names: The identifier strings are required. "gfx0" for graphics layer
@@ -57,6 +61,18 @@ Example:
 
 		power-domains = <&pd_dp>;
 
+		vid-layer {
+			dma-names = "vid0", "vid1", "vid2";
+			dmas = <&xlnx_dpdma 0>,
+			       <&xlnx_dpdma 1>,
+			       <&xlnx_dpdma 2>;
+		};
+
+		gfx-layer {
+			dma-names = "gfx0";
+			dmas = <&xlnx_dpdma 3>;
+		};
+
 		dma-names = "vid0", "vid1", "vid2", "gfx0";
 		dmas = <&xlnx_dpdma 0>,
 		       <&xlnx_dpdma 1>,
-- 
2.31.1

